### ISSUE TITLE
Code Table Request - new part attribute: DNA fragment length

### ISSUE THREAD

**Issue by falco-rk** [ArctosDB/arctos#7395]
### Goal
_Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step._

OGL has a lot of DNA samples and we would like to be able to classify their fragment length or molecular weight as an indicator of DNA quality. OGL (and literature) considers DNA that has fragment lenghts above 10 kb to be high quality, high molecular weight DNA. We usually determine this based on gel electrophoresis data.

### Context
_Describe why this new value is necessary and existing values are not._

Knowing the quality of a DNA sample has implications for its uses.

### Table
_Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure._

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecpart_attribute_type

### Proposed Value
_Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos._

DNA fragment length

### Proposed Definition
_Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable._

DNA fragment length, as determined by gel electrophoresis or other methods, can be used as an indicator of DNA quality

### Collection type
_Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm_

all

#### Attribute data type
_If the request is for an attribute, what values will be allowed?
free-text, categorical, or number+units depending upon the attribute (TBA)_

categorical

#### Attribute controlled values
_If the values are categorical (to be controlled by a code table), add a link to the appropriate code table. If a new table or set of values is needed, please elaborate._

Proposed values:

term | description 
-- | -- 
all DNA above 10kb | all (or most) of the DNA present has fragments lengths greater that 10kb
some DNA above 10kb | some but not all of the DNA present has fragments lengths greater that 10kb
no DNA above 10kb | no DNA has fragments lengths greater that 10kb but DNA less than 10kb is present
no DNA detected | no DNA detected

#### Attribute units
_if numerical values should be accompanied by units, provide a link to the appropriate units table._

#### Part preservation attribute affect on "tissueness"
_if a new part preservation is requested, please add the affect it would have on "tissueness": No Influence, Allows, or Denies_

no influence

### Priority
_Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking._

### Example Data
_Requests with clarifying sample data are generally much easier to understand and prioritize. Please attach or link to any representative data, in any form or format, which might help clarify the request._

### Available for Public View
_Most data are by default publicly available. Describe any necessary access restrictions._

### Helpful Actions

- [ ] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [ ] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators @Jegelewicz @campmlc 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [ ] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [ ] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [ ] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by falco-rk** [comment]
Anyone have any thoughts about this?

**Comment by Jegelewicz** [comment]
I kinda dislike creating a code table for this. Could we just use the yes/no and the thing you have in your data, which I think is MWAbove10kbConsensus?

molecular weight above 10K: NEED TO DEFINE THIS

categorical: yes/no

If I am comparing apples to oranges - let me know!

**Comment by dustymc** [comment]
>creating a code table for this

That's also my hesitation, but I was wondering why not number+units? value=10,000 units=IDK probably something new, method=bla percent-or-whatever above value according to.......

**Comment by Jegelewicz** [comment]
> why not number+units

because these aren't precise....

**Comment by dustymc** [comment]
>aren't precise

method

**Comment by falco-rk** [comment]
Sure, we can do yes/no. I will update things.

>why not number+units

We don't know the exact size, we just know if its larger than 10kb based on a gel (which would be method). Based on our currently methods, we will never know more precisely than greater or less than 10 kb (and for this we don't really care).

**Comment by Jegelewicz** [comment]
@falco-rk I made some tweaks to the definition - let me know if they are crazy.

**Comment by falco-rk** [comment]
Your changes look great!

**Comment by campmlc** [comment]
Could we just call this "high molecular weight" and put the "greater than 10 kb" with the methods? I see a lot of variation in the literature as to what is considered high molecular weight DNA, and that way we don't constrain this to a particular method or value, as different labs may have different cutoffs and methods?

**Comment by Jegelewicz** [comment]
@falco-rk see above.

**Comment by falco-rk** [comment]
>Could we just call this "high molecular weight" and put the "greater than 10 kb" with the methods?

No. My major problem with that is we need to be able to indicate that DNA is _not_ high molecular weight/less than 10kb. If its just called "high molecular weight," there isn't a way to say that we have tested it in some way and have data to say it is low quality. And we can't assume that the DNA is not high molecular weight if that attribute is missing because we don't have that data for every DNA extract.

I also don't love that putting "greater than 10kb" in methods is an additional step and doesn't have any controlled vocabulary. I know there is a lot of variation in the literature and 10kb is what we have chosen but I want to make sure our data remains clear.

**Comment by campmlc** [comment]
It sounds like you need 2 part attributes, then?  One for high MW and one for low MW, to indicate that the sample was indeed checked and the low MW value wasn't just missing or not entered?
I'm just concerned that another collection may use a different cutoff, but we could cross that bridge once that happens, and @dustymc could automagic the 10kb value into your methods at that time.

**Comment by falco-rk** [comment]
>It sounds like you need 2 part attributes, then?

No, you could say "no" to molecular weight above 10Kb which would indicate that the sample was checked and there is not DNA greater than 10kb.
